custom/abt/reports/expressions.py

```diff
@@ -113,9 +113,9 @@ def _get_language(cls, item):
 
     @classmethod
     def _get_warning(cls, spec, item):
-        default = spec.get("warning", "")
+        default = unicode(spec.get("warning", u""))
         if cls._get_language(item) == "fra":
-            return spec.get("french", default)
+            return unicode(spec.get("french", default))
```

Comment:
@millerdev 
Yeah, so the problem was actually down on [line 180](https://github.com/dimagi/commcare-hq/pull/6833/files#diff-de93bf3c69284ffa146c1710094c821aL180). `unchecked` is a list of unicode strings, and if one of them has a non ascii character, then I get an `UnicodeEncodeError` from the `format` method.

So, my solution was to make sure that `_get_warning` always returns a unicode string. The `spec` argument is parsed yaml, so sometimes `spec.get("warning")` and  `spec.get("french")` are `str` objects (if the string in the `.yaml` file didn't contain non-ascii characters), and sometimes they are `unicode` objects (if the string in the `.yaml` file did contain non-ascii characters).

Does this seem like the right approach, or would you recommend something else?